packages/agreement-process/src/services/agreementActivationProcessor.ts

```diff
@@ -48,75 +40,43 @@ import {
 } from "./agreementStateProcessor.js";
 import { contractBuilder } from "./agreementContractBuilder.js";
 import { AgreementQuery } from "./readmodel/agreementQuery.js";
-import { EserviceQuery } from "./readmodel/eserviceQuery.js";
 import { TenantQuery } from "./readmodel/tenantQuery.js";
 import {
   createStamp,
   suspendedByConsumerStamp,
   suspendedByProducerStamp,
 } from "./agreementStampUtils.js";
 import { AttributeQuery } from "./readmodel/attributeQuery.js";
+import { retrieveTenant } from "./agreementService.js";
 
-export async function activateAgreementLogic(
-  agreementId: AgreementId,
-  agreementQuery: AgreementQuery,
-  eserviceQuery: EserviceQuery,
-  tenantQuery: TenantQuery,
-  attributeQuery: AttributeQuery,
-  authData: AuthData,
-  storeFile: FileManager["storeBytes"],
-  correlationId: string,
-  logger: Logger
-): Promise<[Agreement, Array<CreateEvent<AgreementEvent>>]> {
-  const agreement = await agreementQuery.getAgreementById(agreementId);
-  assertAgreementExist(agreementId, agreement);
-
-  assertRequesterIsConsumerOrProducer(agreement.data, authData);
-  verifyConsumerDoesNotActivatePending(agreement.data, authData);
-  assertActivableState(agreement.data);
-
-  const eservice = await eserviceQuery.getEServiceById(
-    agreement.data.eserviceId
-  );
-  assertEServiceExist(agreement.data.eserviceId, eservice);
+export async function processActivateAgreement({
+  agreementData,
+  eservice,
+  authData,
+  tenantQuery,
+  agreementQuery,
+  attributeQuery,
+  storeFile,
+  correlationId,
+  logger,
+}: {
+  agreementData: WithMetadata<Agreement>;
+  eservice: EService;
+  authData: AuthData;
+  tenantQuery: TenantQuery;
+  agreementQuery: AgreementQuery;
+  attributeQuery: AttributeQuery;
+  storeFile: FileManager["storeBytes"];
+  correlationId: string;
+  logger: Logger;
+}): Promise<[Agreement, Array<CreateEvent<AgreementEvent>>]> {
+  const agreement = agreementData.data;
 
+  const consumer = await retrieveTenant(agreement.consumerId, tenantQuery);
```

Comment:
Same question I made in the other processor: why are other entities (Eservices, Agreements) retrieved in `agreementService.ts` while Tenants (producer/consumer) are retrieved here in the processor?

Reply:
> Same question I made in the other processor: why are other entities (Eservices, Agreements) retrieved in `agreementService.ts` while Tenants (producer/consumer) are retrieved here in the processor?

See my comment [here](https://github.com/pagopa/interop-be-monorepo/pull/500#discussion_r1601287412)